Sorts/MergeSort.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @function mergeSort
+ * @description keeps on dividing the list into equal halves until it can no more be divided. By definition, if it is only one element in the list, it is sorted.
+ * @see [Merge Sort](https://www.javatpoint.com/merge-sort)
+ * @example MergeSort([8, 3, 5, 1, 4, 2]) = [1, 2, 3, 4, 5, 8]
+ */
+
+ export const MergeSort = (items: number[]): number[] => {
+    var halfLength = Math.ceil(items.length / 2);
+    var low = items.slice(0, halfLength);
+    var high = items.slice(halfLength);
+    if (halfLength > 1) {
```

Comment:
Is this check necessary? `MergeSort` should properly deal with empty halves.

Reply:
Yes it is necessary, otherwise the stack limit will exceed.